src/main/groovy/org/xbib/gradle/plugin/JFlexTask.groovy

```diff
@@ -21,53 +28,90 @@ import org.gradle.api.tasks.TaskAction
+
+    @Input
+    abstract Property<Boolean> getTime()
+
+    @Input
+    abstract Property<Boolean> getDot()
+
+    @Input
+    abstract Property<Boolean> getDump()
+
+    @Input
+    abstract Property<Boolean> getLegacy_dot()
+
+    @Input
+    abstract Property<Boolean> getStatistics()
+
+    @Input
+    abstract Property<Boolean> getWriteIntoJavaSrc()
+
     @InputFiles
     @PathSensitive(PathSensitivity.RELATIVE)
     Iterable<File> source
 
     @OutputDirectory
-    File target
-
-    @Internal
-    SourceSet theSourceSet
+    abstract DirectoryProperty getTarget()
 
     @TaskAction
     void generateAndTransformJflex() throws Exception {
-        JFlexExtension ext = project.extensions.findByType(JFlexExtension)
-        Options.setRootDirectory(ext.rootDirectory ? ext.rootDirectory : new File(""))
+        Options.setRootDirectory(rootDirectory.asFile.getOrElse(new File("")))
         Skeleton.readDefault()
-        if (ext.skel) {
-            Skeleton.readSkelFile(ext.skel)
-        }
-        Options.encoding = ext.encoding ? Charset.forName(ext.encoding) : Charset.defaultCharset()
-        Options.verbose = ext.verbose
-        Options.progress = ext.progress
-        Options.unused_warning = ext.unused_warning
-        Options.jlex = ext.jlex
-        Options.no_minimize = ext.no_minimize
-        Options.no_backup = ext.no_backup
-        Options.time = ext.time
-        Options.dot = ext.dot
-        Options.dump = ext.dump
-        Options.legacy_dot = ext.legacy_dot
-        // hack for writing directly into java source. Not recommended.
-        if (ext.writeIntoJavaSrc) {
-            if (theSourceSet.java && theSourceSet.java.srcDirs) {
-                logger.info "java sources: ${theSourceSet.java.srcDirs}"
-                target = theSourceSet.java.srcDirs.first()
-                logger.info "switching to first java source directory ${target}"
-            } else {
-                logger.warn "writing into java source not possible, is empty"
-            }
+        if (skel.isPresent()) {
+            Skeleton.readSkelFile(skel.get().asFile)
         }
+        Options.encoding = encoding.map({ Charset.forName(it) }).getOrElse(Charset.defaultCharset())
+        Options.verbose = verbose.get()
+        Options.progress = progress.get()
+        Options.unused_warning = unused_warning.get()
+        Options.jlex = jlex.get()
+        Options.no_minimize = no_minimize.get()
+        Options.no_backup = no_backup.get()
+        Options.time = time.get()
+        Options.dot = dot.get()
+        Options.dump = dump.get()
+        Options.legacy_dot = legacy_dot.get()
+        File targetFile = target.get().asFile
         source.each { file ->
             String pkg = getPackageName(file)
-            File fullTarget = new File(target, pkg.replace('.','/'))
-            project.mkdir(fullTarget)
+            File fullTarget = new File(targetFile, pkg.replace('.', '/'))
+            fullTarget.mkdirs()
```

Comment:
project.mkdir is replaced with usage of java API based on recommendations from: https://docs.gradle.org/7.3.2/userguide/configuration_cache.html#config_cache:requirements:disallowed_types